lib/livebook/application.ex

```diff
@@ -223,35 +226,47 @@ defmodule Livebook.Application do
     end
   end
 
-  defp create_teams_hub() do
+  defp parse_teams_hub() do
     teams_key = System.get_env("LIVEBOOK_TEAMS_KEY")
     auth = System.get_env("LIVEBOOK_TEAMS_AUTH")
 
     cond do
       teams_key && auth ->
-        Application.put_env(:livebook, :teams_auth?, true)
-
-        hub =
+        fun =
           case String.split(auth, ":") do
             ["offline", name, public_key] ->
-              create_offline_hub(teams_key, name, public_key)
+              Application.put_env(:livebook, :teams_auth, :offline)
+              fn -> create_offline_hub(teams_key, name, public_key) end
 
             ["online", name, org_id, org_key_id, agent_key] ->
-              create_online_hub(teams_key, name, org_id, org_key_id, agent_key)
+              Application.put_env(:livebook, :teams_auth, :online)
+              fn -> create_online_hub(teams_key, name, org_id, org_key_id, agent_key) end
 
             _ ->
               Livebook.Config.abort!("Invalid LIVEBOOK_TEAMS_AUTH configuration.")
           end
 
-        Application.put_env(:livebook, :apps_path_hub_id, hub.id)
+        case Application.fetch_env(:livebook, :identity_provider) do
```

Comment:
Let's move this to the online branch and do this:

```
        with :error <- Application.fetch_env(:livebook, :identity_provider) do
            Application.put_env(
              :livebook,
              :identity_provider,
              {:zta, Livebook.ZTA.LivebookTeams, hub_id}
            )
        end
```

This way we get rid of :teams_auth_hub_id too.